### PR TITLE
Bluespace Artillery now requires keycard authentication to unlock controls

### PR DESCRIPTION
**Pull request by silicons**
This is on @KorPhaeron's planetstation list.

### CODE REVIEW

File: tgui/src/interfaces/bsa.ract [hunk 1]
```diff
@@ -9,7 +9,13 @@
     <ui-button icon='crosshairs' action='recalibrate'>{{data.target}}</ui-button>
   </ui-section>
   <ui-section label='Controls'>
-    <ui-button icon='warning' state='{{data.ready ? null : "disabled"}}' action='fire'>FIRE!</ui-button>
+	{{#if !data.unlocked}}
+		<ui-notice>
+			<span>Bluespace Artillery firing protocals must be globally unlocked from two keycard authentication devices first!</span>
```
Comment (Cyberboss) [inline tgui/src/interfaces/bsa.ract]: >protocals

File: code/modules/security_levels/keycard_authentication.dm [hunk 2]
```diff
@@ -117,3 +124,7 @@ GLOBAL_VAR_INIT(emergency_access, FALSE)
 			D.update_icon(0)
 	minor_announce("Access restrictions in maintenance areas have been restored.", "Attention! Station-wide emergency rescinded:")
 	GLOB.emergency_access = FALSE
+
+/proc/toggle_bluespace_artillery()
+	GLOB.bsa_unlock = !GLOB.bsa_unlock
+	minor_announce("Bluespace Artillery firing protocals have been [GLOB.bsa_unlock? "unlocked" : "locked"]", "ATTENTION: Weapons Systems Update:")
```
Comment (Cyberboss) [inline code/modules/security_levels/keycard_authentication.dm]: >protocals

Comment (Cyberboss) [inline code/modules/security_levels/keycard_authentication.dm]: Remove `ATTENTION:` and the colon after Weapons Systems Update

File: code/modules/security_levels/keycard_authentication.dm [hunk 3]
```diff
@@ -99,7 +104,9 @@ GLOBAL_DATUM_INIT(keycard_events, /datum/events, new)
 		if("Emergency Maintenance Access")
 			make_maint_all_access()
 			SSblackbox.inc("alert_keycard_auth_maint",1)
-
+		if("Bluespace Artillery Unlock")
+			toggle_bluespace_artillery()
+			SSblackbox.inc("alert_keycard_auth_bsa",1)
```
Comment (Cyberboss) [inline code/modules/security_levels/keycard_authentication.dm]: Remove the second param, it's 1 by default

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: tgui/src/interfaces/bsa.ract (hunk 1 above)
Comment (octareenroon91) [inline tgui/src/interfaces/bsa.ract]: courtesy: "protocols"

**Comment by octareenroon91** [comment]
I suggest the BSA deployment spawn one, possibly two, keycard auth devices.

File: tgui/src/interfaces/bsa.ract (hunk 1 above)
Comment (lzimann) [inline tgui/src/interfaces/bsa.ract]: I don't like the `must be globallly unlocked` part, doesn't mean much in the game context.

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Recompile tgui, you spoon!

**Review by Cyberboss — APPROVED** [review]

**Comment by PraiseRatvar** [comment]
hey look its technically actual planetstation progress

**Pull request merged**